cli/src/complete.rs

```diff
@@ -43,6 +43,24 @@ pub fn local_bookmarks() -> Vec<CompletionCandidate> {
     })
 }
 
+/// A specific completion function for the `new` argument of `bookmark rename`.
+/// It suggests the `old` argument, because the user may want to fix a typo.
+pub fn bookmark_rename_new() -> Vec<CompletionCandidate> {
+    let mut candidates = Vec::new();
+
+    let bookmark_to_rename = std::env::args_os()
+        .rev()
+        .filter_map(|arg| arg.into_string().ok())
+        .filter(|arg| !arg.starts_with("-"))
+        // The first non-flag argument is the one being completed, the second
+        // one is the "old" argument.
+        .nth(1);
+    if let Some(bookmark) = bookmark_to_rename {
+        candidates.push(CompletionCandidate::new(bookmark));
+    }
```

Comment:
Isn't it flaky if there are global argument after `<OLD_NAME>` for example?

I personally don't think this would be worth the effort. Maybe better to revisit after clap add something like https://github.com/clap-rs/clap/issues/5784?

Reply:
Yeah, makes sense.